### PR TITLE
align H3 streaming request behavior to that of H2

### PR DESCRIPTION
**Pull request by sleepybishop**
This pr ...
  - renames `H2O_HTTP2_SETTINGS_HOST_MAX_CONCURRENT_STREAMING_REQUESTS` to `H2O_HTTP2_DEFAULT_MAX_CONCURRENT_STREAMING_REQUESTS` and adds a similar declaration to H3.
  - sets `H2O_HTTP3_REQUEST_BODY_MIN_BYTES_TO_BLOCK` to 1 to have parity with H2.
  - introduces `http3-max-concurrent-streaming-requests-per-connection` to have parity with H2.

### CODE REVIEW

**Comment by sleepybishop** [comment]
The intention behind the previous 256 limit check came from the docs for http2 max concurrent streams, the docs say the max is 256 but I didn't see where that was enforced. I removed the extra check in my changes to maintain parity with h2. Let me know if you would like to update the checks and/or docs.

File: srcdoc/configure/http3_directives.mt [hunk 1]
```diff
@@ -124,6 +124,22 @@ See <a href="configure/http2_directives.html#http2-input-window-size"><code>http
...
+    default => 'http3-max-concurrent-streaming-requests-per-connection: 1',
+    desc    => <<'EOT',
+Maximum number of streaming requests to be handled concurrently within a single HTTP/3 connection.
+EOT
+)->(sub {
+?>
+<p>
+The value cannot exceed 256.
+</p>
+? })
```
Comment (kazuho) [inline srcdoc/configure/http3_directives.mt:140]:
```suggestion
)->(sub {});
```

Let's remove this.

Maximum concurrency of 256 is an old limit in the HTTP/2 code, I doubt we have it any more, but moreover, there is no such restriction in our H3 code.

**Review by kazuho — COMMENTED** [review]
Wonderful! Thank you for the PR.

I think we can just drop the 256 restriction (see my comments below), that all I want to say.

**Comment by kazuho** [comment]
Wonderful! Merged.

**Pull request merged**